src/libs/state.ts

```diff
@@ -4,12 +4,72 @@ import type { WorkspaceWrapper } from 'src/libs/workspace-utils';
 
 export const routeHandlersStore = atom<unknown[]>([]);
 
-export const authStore = atom<any>({
+export type TerraUser = {
+  token?: string | undefined;
+  scope?: string | undefined;
+  id?: string | undefined;
+  email?: string | undefined;
+  name?: string | undefined;
+  givenName?: string | undefined;
+  familyName?: string | undefined;
+  imageUrl?: string | undefined;
+  idp?: string | undefined;
+};
+
+export type TerraUserProfile = {
+  institute: string | undefined;
+  title: string | undefined;
+  department: string | undefined;
+  interestInTerra: string | undefined;
+};
+
+export type TokenMetadata = {
+  token: string | undefined; // do not log or send this to mixpanel
+  id: string | undefined;
+  createdAt: number;
+  expiresAt: number;
+  totalTokensUsedThisSession: number;
+  totalTokenLoadAttemptsThisSession: number;
+};
+
+export type AuthState = {
+  anonymousId: string | undefined;
+  authContext: any;
+  authTokenMetadata: TokenMetadata;
+  cookiesAccepted: boolean | undefined;
+  fenceStatus: {};
+  hasGcpBillingScopeThroughB2C: boolean | undefined;
+  isAzurePreviewUser?: boolean | undefined;
+  isSignedIn: boolean | undefined;
+  isTimeoutEnabled?: boolean | undefined;
+  nihStatus?: {
+    linkedNihUsername: string;
+    linkExpireTime: number;
+  };
+  oidcConfig: {
+    authorityEndpoint?: string;
+    clientId?: string;
+  };
+  // props in the TerraUserProfile are always present, but there may be more props
+  profile: TerraUserProfile & any;
```

Comment:
Let's include the arbitrary keys in the `TerraUserProfile` type instead of here.

Also, I think we could do better than `any` with something like:
```ts
export type TerraUserProfile = {
  institute: string | undefined;
  title: string | undefined;
  department: string | undefined;
  interestInTerra: string | undefined;
} & {
  [key: string]: string
};
```